diff --git a/node_modules/@sominaththore/react-native-custom-status-bar/CustomStatusBar.js b/node_modules/@sominaththore/react-native-custom-status-bar/CustomStatusBar.js
index 803faeb..2c9b291 100644
--- a/node_modules/@sominaththore/react-native-custom-status-bar/CustomStatusBar.js
+++ b/node_modules/@sominaththore/react-native-custom-status-bar/CustomStatusBar.js
@@ -1,23 +1,61 @@
 import React from 'react';
-import { SafeAreaView, Platform, StatusBar, StyleSheet } from 'react-native';
-import { getStatusBarHeight } from './status-bar-height/index.js';
+import {
+    Platform,
+    StatusBar,
+    StyleSheet,
+    View,
+} from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
 
 const CustomStatusBar = ({ backgroundColor, barStyle }) => {
+    const insets = useSafeAreaInsets();
+
+    if (Platform.OS === 'android') {
+        return (
+            <>
+                <View
+                    pointerEvents="none"
+                    style={[
+                        styles.androidStatusBar,
+                        {
+                            height: insets.top,
+                            backgroundColor,
+                        },
+                    ]}
+                />
+
+                <StatusBar barStyle={barStyle} />
+            </>
+        );
+    }
+
     return (
-        <SafeAreaView style={[styles.statusbar, { backgroundColor, height: getStatusBarHeight() }]}>
-            <StatusBar
-                translucent
-                backgroundColor={backgroundColor}
-                barStyle={barStyle}
-            />
-        </SafeAreaView>
-    )
-}
+        <View
+            style={[
+                styles.iosStatusBar,
+                {
+                    height: insets.top,
+                    backgroundColor,
+                },
+            ]}
+        >
+            <StatusBar barStyle={barStyle} />
+        </View>
+    );
+};
 
 const styles = StyleSheet.create({
-    statusbar: {
-        height: Platform.OS === 'ios' ? getStatusBarHeight() : 0
-    }
-})
+    androidStatusBar: {
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        right: 0,
+        zIndex: 9999,
+    },
+
+    iosStatusBar: {
+        width: '100%',
+    },
+});
 
 export default CustomStatusBar;
\ No newline at end of file
